pkg/client/client.go

```diff
@@ -93,36 +102,46 @@ func NewClient(gvr schema.GroupVersionResource, kind string, namespaced bool, cl
 
 func noop() {}
 
-func (c *Client) setupCtx(ctx context.Context, minTimeout time.Duration) (context.Context, func()) {
-	if minTimeout == 0 && c.timeout == 0 {
+// setupCtx wraps the provided context with client.timeout, and returns the new context and it's cancel func.
+// If client.timeout is 0 then the provided context is returned with a noop function instead of a cancel function.
+func (c *Client) setupCtx(ctx context.Context) (context.Context, func()) {
+	if c.timeout == 0 {
 		return ctx, noop
 	}
 
-	timeout := c.timeout
-	if minTimeout > 0 && timeout < minTimeout {
-		timeout = minTimeout
-	}
-
-	return context.WithTimeout(ctx, timeout)
+	return context.WithTimeout(ctx, c.timeout)
 }
 
-func (c *Client) Get(ctx context.Context, namespace, name string, result runtime.Object, options metav1.GetOptions) (err error) {
+// Get will attempt to find the requested resource with the given name in the given namespace (if client.Namespaced is set to true).
+// Get will then attempt to unmarshal the response into the provide result object.
+// If the returned response object is of type Status and has .Status != StatusSuccess, the
+// additional information in Status will be used to enrich the error.
+func (c *Client) Get(ctx context.Context, namespace, name string, result runtime.Object, opts GetOptions) error {
 	defer c.setKind(result)
-	ctx, cancel := c.setupCtx(ctx, 0)
+	ctx, cancel := c.setupCtx(ctx)
 	defer cancel()
-	err = c.RESTClient.Get().
+	req := c.RESTClient.Get().
 		Prefix(c.prefix...).
 		NamespaceIfScoped(namespace, c.Namespaced).
 		Resource(c.resource).
 		Name(name).
-		VersionedParams(&options, metav1.ParameterCodec).
-		Do(ctx).
-		Into(result)
-	return
+		VersionedParams(&opts.GetOptions, metav1.ParameterCodec)
+
+	for _, modifier := range opts.RequestModifiers {
```

Comment:
Nit:

Might be good to expose a `opts.ApplyModifiers(request)` function so you can avoid doing a for loop each time. It would also allow you to keep the current structure of chained function calls.

Reply:
I added an `apply` function though this does not allow us to chain function calls since we can not add methods to the `rest.Request` struct.